### PR TITLE
Use common OIDC token env vars for identity live tests

### PR DESCRIPTION
**Pull request by benbp**
Support oidc az cli login in post deploy scripts for multi-cloud. Requires https://github.com/Azure/azure-sdk-tools/pull/9099

### CODE REVIEW

File: sdk/identity/test-resources-pre.ps1 [hunk 1]
```diff
@@ -23,9 +24,11 @@ $sshKey = Get-Content $PSScriptRoot/sshKey.pub
 
 $templateFileParameters['sshPubKey'] = $sshKey
 
-# Get the max version that is not preview and then get the name of the patch version with the max value
+az cloud set --name $Environment
 az login --service-principal -u $TestApplicationId --tenant $TenantId --allow-no-subscriptions --federated-token $env:ARM_OIDC_TOKEN
-$versions = az aks get-versions -l westus -o json | ConvertFrom-Json
+# Get the max version that is not preview and then get the name of the patch version with the max value
+$region = if ($Environment -eq 'AzureUSGovernment') { 'usgovvirginia' } elseif ($Environment -eq 'AzureChinaCloud') { 'chinanorth3' } else { 'westus' }
```
Comment (weshaggard) [inline sdk/identity/test-resources-pre.ps1:30]: Seems like something else we might want to export at some point.

**Review by weshaggard — APPROVED** [review]

File: sdk/identity/test-resources-pre.ps1 (hunk 1 above)
Comment (benbp) [reply sdk/identity/test-resources-pre.ps1:30]: Ideally we pass `Location` as a parameter down to the pre/post scripts the same way we do Environment. I checked but don't think I saw a place where we are doing it?

**Review by christothes — APPROVED** [review]
LGTM - can we validate the latest commits with `/azp run net - identity - tests` ?

**Comment by benbp** [comment]
@christothes I've been running the weekly in the background. Trying to find the right VM sku that will have capacity in our TME sub right now (my latest change to v5 didn't work).

**Pull request closed without merging**